url901cc5721b_shidasan_KonohaCoffee_tgz-pJ8-org_KonohaScript_KonohaJ8/src/org/KonohaScript/Konoha.java

```diff
@@ -0,0 +1,253 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.KonohaScript.GrammarSet.MiniKonoha;
+
+/* konoha util */
+
+
+
+// Runtime
+
+// #define kContext_Debug ((khalfflag_t)(1<<0))
+// #define kContext_Interactive ((khalfflag_t)(1<<1))
+// #define kContext_CompileOnly ((khalfflag_t)(1<<2))
+// #define kContext_Test ((khalfflag_t)(1<<3))
+// #define kContext_Trace ((khalfflag_t)(1<<4))
+//
+// #define KonohaContext_Is(P, X) (KFlag_Is(khalfflag_t,(X)->stack->flag,
+// kContext_##P))
+// #define KonohaContext_Set(P, X) KFlag_Set1(khalfflag_t, (X)->stack->flag,
+// kContext_##P)
+
+class KKeyIdMap {
+	int GetId(String key) {
+		return 0;
+	}
+
+	void SetId(String key, int id) {
+
+	}
+}
+
+class KParamMap {
+	int GetId(int hash) {
+		return 0;
+	}
+
+	void SetId(int hash, int id) {
+
+	}
+}
+
+class KSymbolTable implements KonohaParserConst {
+	ArrayList<KClass> ClassList;
+	HashMap<String, KClass> ClassNameMap;
+
+	ArrayList<KPackage> PackageList;
+	KKeyIdMap PackageMap;
+
+	ArrayList<String> FileIdList;
+	HashMap<String,Integer> FileIdMap;
+	
+	ArrayList<String> SymbolList;
+	HashMap<String,Integer> SymbolMap;
+	
+	ArrayList<KParam> ParamList;
+	KParamMap ParamMap;
+	ArrayList<KParam> SignatureList;
+	KParamMap SignatureMap;
+
+	KSymbolTable() {
+		this.ClassList = new ArrayList<KClass>(64);
+		this.ClassNameMap = new HashMap<String, KClass>();
+		
+		this.FileIdList = new ArrayList<String>(16);
+		this.FileIdMap = new HashMap<String, Integer>();
+
+		this.SymbolList = new ArrayList<String>(64);
+		this.SymbolMap = new HashMap<String, Integer>();
+
+		this.PackageList = new ArrayList<KPackage>(16);
+		this.ParamList = new ArrayList<KParam>(64);
+		this.SignatureList = new ArrayList<KParam>(64);
+		this.PackageMap = new KKeyIdMap();
+		this.ParamMap = new KParamMap();
+		this.SignatureMap = new KParamMap();
+	}
+
+	void Init(Konoha kctx) {
+		KPackage defaultPackage = NewPackage(kctx, "Konoha");
+//		NewClass(kctx, defaultPackage, "void");
+	}
+
+	long GetFileId(String file, int linenum) {
+		Integer fileid = this.FileIdMap.get(file);
+		if (fileid == null) {
+			int id = FileIdList.size();
+			FileIdList.add(file);
+			FileIdMap.put(file, new Integer(id));
+			return ((long)id << 32) | linenum;
+		}
+		return (fileid.longValue() << 32) | linenum;
+	}
+
+	String GetFileName(long uline) {
+		int id = (int)(uline >> 32);
+		return FileIdList.get(id);
+	}
+	
+	// Symbol 
+	
+	public final static int MaskSymbol(int n, int mask) {
+		return (n << 2) | mask;
+	}
+
+	public final static int UnmaskSymbol(int id) {
+		return id >> 2;
+	}
+
+	public String StringfySymbol(int Symbol) {
+		String key = SymbolList.get(UnmaskSymbol(Symbol));
+		if((Symbol & GetterSymbol) == GetterSymbol) {
+			return "Get" + key;
+		}
+		if((Symbol & SetterSymbol) == SetterSymbol) {
+			return "Get" + key;
+		}
+		if((Symbol & MetaSymbol) == MetaSymbol) {
+			return "\\" + key;
+		}
+		return key;
+	}
+
+	public int GetSymbol(String Symbol, int DefaultValue) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 150-160 into a separate method.